From f8d0d4b84c90cb124a21b1faa6d538b80b846c41 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@gmail.com>
Date: Tue, 29 Sep 2026 10:03:20 +0200
Subject: [PATCH] bytemuck: fix build with rust-1.95

See:
https://github.com/void-linux/void-packages/pull/60078

Avoid:

[14991/46974] RUST obj/third_party/rust/bytemuck/v1/lib/libbytemuck_lib.rlib obj/third_party/rust/bytemuck/v1/lib/libbytemuck_lib.rmeta obj/third_party/rust/bytemuck/v1/lib.rustflags.json obj/third_party/rust/bytemuck/v1/lib.rsp.rust
FAILED: obj/third_party/rust/bytemuck/v1/lib/libbytemuck_lib.rlib obj/third_party/rust/bytemuck/v1/lib/libbytemuck_lib.rmeta obj/third_party/rust/bytemuck/v1/lib.rustflags.json obj/third_party/rust/bytemuck/v1/lib.rsp.rust
"/OE/build/luneos-blacksail/tmp/work/corei7-64-webos-linux/webruntime-clang/151.0.7922.108-0/recipe-sysroot-native/usr/bin/python3-native/python3" "../../sources/webruntime-clang-151.0.7922.108-0/src/build/rust/gni_impl/rustc_wrapper.py" --dump-rustc-env-and-flags=obj/third_party/rust/bytemuck/v1/lib.rustflags.json --rsp=obj/third_party/rust/bytemuck/v1/lib.rsp --rustc=../../recipe-sysroot-native/usr/bin/rustc --depfile=obj/third_party/rust/bytemuck/v1/lib/libbytemuck_lib.rlib.d --emit=dep-info=obj/third_party/rust/bytemuck/v1/lib/libbytemuck_lib.rlib.d,link,metadata=obj/third_party/rust/bytemuck/v1/lib/libbytemuck_lib.rmeta -o obj/third_party/rust/bytemuck/v1/lib/libbytemuck_lib.rlib -- -Clinker="/OE/build/luneos-blacksail/tmp/work/corei7-64-webos-linux/webruntime-clang/151.0.7922.108-0/recipe-sysroot-native/usr/bin/clang++" ../../sources/webruntime-clang-151.0.7922.108-0/src/third_party/rust/chromium_crates_io/vendor/bytemuck-v1/src/lib.rs LDFLAGS RUSTENV OUT_DIR=gen/third_party/rust/bytemuck/v1/lib CARGO_PKG_AUTHORS=Lokathor\ \<zefria@gmail.com\> CARGO_PKG_VERSION=1.25.0 CARGO_PKG_VERSION_MAJOR=1 CARGO_PKG_VERSION_MINOR=25 CARGO_PKG_VERSION_PATCH=0 CARGO_PKG_VERSION_PRE= CARGO_PKG_NAME=bytemuck CARGO_PKG_DESCRIPTION=A\ crate\ for\ mucking\ around\ with\ piles\ of\ bytes. CARGO_PKG_REPOSITORY=https://github.com/Lokathor/bytemuck CARGO_MANIFEST_DIR=../../sources/webruntime-clang-151.0.7922.108-0/src/third_party/rust/bytemuck/v1/lib/crate
error[E0425]: cannot find type `LaneCount` in module `core::simd`
   --> ../../sources/webruntime-clang-151.0.7922.108-0/src/third_party/rust/chromium_crates_io/vendor/bytemuck-v1/src/zeroable.rs:234:15
    |
234 |   core::simd::LaneCount<N>: core::simd::SupportedLaneCount,
    |               ^^^^^^^^^ not found in `core::simd`

error[E0405]: cannot find trait `SupportedLaneCount` in module `core::simd`
   --> ../../sources/webruntime-clang-151.0.7922.108-0/src/third_party/rust/chromium_crates_io/vendor/bytemuck-v1/src/zeroable.rs:234:41
    |
234 |   core::simd::LaneCount<N>: core::simd::SupportedLaneCount,
    |                                         ^^^^^^^^^^^^^^^^^^ not found in `core::simd`

error[E0425]: cannot find type `LaneCount` in module `core::simd`
   --> ../../sources/webruntime-clang-151.0.7922.108-0/src/third_party/rust/chromium_crates_io/vendor/bytemuck-v1/src/pod.rs:155:15
    |
155 |   core::simd::LaneCount<N>: core::simd::SupportedLaneCount,
    |               ^^^^^^^^^ not found in `core::simd`

error[E0405]: cannot find trait `SupportedLaneCount` in module `core::simd`
   --> ../../sources/webruntime-clang-151.0.7922.108-0/src/third_party/rust/chromium_crates_io/vendor/bytemuck-v1/src/pod.rs:155:41
    |
155 |   core::simd::LaneCount<N>: core::simd::SupportedLaneCount,
    |                                         ^^^^^^^^^^^^^^^^^^ not found in `core::simd`

error: aborting due to 4 previous errors

Some errors have detailed explanations: E0405, E0425.
For more information about an error, try `rustc --explain E0405`.
ERROR: `rustc` exited with a non-zero exit code: 1

Upstream-Status: Pending
Signed-off-by: Martin Jansa <martin.jansa@gmail.com>
---
 .../rust/chromium_crates_io/vendor/bytemuck-v1/src/pod.rs        | 1 -
 .../rust/chromium_crates_io/vendor/bytemuck-v1/src/zeroable.rs   | 1 -
 2 files changed, 2 deletions(-)

diff --git a/src/third_party/rust/chromium_crates_io/vendor/bytemuck-v1/src/pod.rs b/src/third_party/rust/chromium_crates_io/vendor/bytemuck-v1/src/pod.rs
index 330f722b34..b0397923c7 100644
--- a/src/third_party/rust/chromium_crates_io/vendor/bytemuck-v1/src/pod.rs
+++ b/src/third_party/rust/chromium_crates_io/vendor/bytemuck-v1/src/pod.rs
@@ -152,7 +152,6 @@ impl_unsafe_marker_for_simd!(
 unsafe impl<T, const N: usize> Pod for core::simd::Simd<T, N>
 where
   T: core::simd::SimdElement + Pod,
-  core::simd::LaneCount<N>: core::simd::SupportedLaneCount,
 {
 }
 
diff --git a/src/third_party/rust/chromium_crates_io/vendor/bytemuck-v1/src/zeroable.rs b/src/third_party/rust/chromium_crates_io/vendor/bytemuck-v1/src/zeroable.rs
index 186c567fff..397dddec99 100644
--- a/src/third_party/rust/chromium_crates_io/vendor/bytemuck-v1/src/zeroable.rs
+++ b/src/third_party/rust/chromium_crates_io/vendor/bytemuck-v1/src/zeroable.rs
@@ -231,7 +231,6 @@ impl_unsafe_marker_for_simd!(
 unsafe impl<T, const N: usize> Zeroable for core::simd::Simd<T, N>
 where
   T: core::simd::SimdElement + Zeroable,
-  core::simd::LaneCount<N>: core::simd::SupportedLaneCount,
 {
 }
 
